File: kotlin/src/main/kotlin/com/joshy/hywrapper/model/skyblock/profile/Bestiary.kt
package com.joshy.hywrapper.model.skyblock.profile

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Bestiary(
    @SerialName("migrated_stats")
    val migratedStats: Boolean? = null,
    @SerialName("migration")
    val migration: Boolean? = null,
    val kills: Map<String, Int>? = null, // TODO enum
    val deaths: Map<String, Int>? = null, // TODO enum
    val milestone: Milestone? = null,
    val miscellaneous: Miscellaneous? = null
)

@Serializable
data class Milestone(
    @SerialName("last_claimed_milestone")
    val lastClaimedMilestone: Int? = null,
)

@Serializable
data class Miscellaneous(
    @SerialName("max_kills_visible")
    val maxKillsVisible: Boolean? = null,
    @SerialName("milestones_notifications")
    val milestonesNotifications: Boolean? = null,
)

File: typescript/src/models/skyblock/profile/Bestiary.ts
export interface Bestiary {
    migrated_stats?: boolean;
    migration?: boolean;
    /** TODO enum */
    kills?: Record<string, number>;
    /** TODO enum */
    deaths?: Record<string, number>;
    milestone?: Milestone;
    miscellaneous?: Miscellaneous;
}

export interface Milestone {
    last_claimed_milestone?: number;
}

export interface Miscellaneous {
    max_kills_visible?: boolean;
    milestones_notifications?: boolean;
}
